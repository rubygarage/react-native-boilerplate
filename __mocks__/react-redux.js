import * as reactRedux from 'react-redux';

export const dispatch = jest.fn();

export const useDispatch = jest.fn(() => dispatch);
export const useSelector = jest.fn();
export const { Provider } = reactRedux;
export const { connect } = reactRedux;
