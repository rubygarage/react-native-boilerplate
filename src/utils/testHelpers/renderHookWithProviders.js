/* eslint-disable import/no-extraneous-dependencies */
import { renderHook } from '@testing-library/react-hooks';
import { ThemeProvider } from 'styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import mockedTheme from 'utils/testHelpers/mockedTheme';

const store = configureStore()({});
store.dispatch = jest.fn();

const renderHookWithProviders = (hook) => {
  const wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={mockedTheme}>
        {children}
      </ThemeProvider>
    </Provider>
  );

  return renderHook(hook, { wrapper });
};

export default renderHookWithProviders;
