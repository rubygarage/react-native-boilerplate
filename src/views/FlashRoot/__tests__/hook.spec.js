import { act } from '@testing-library/react-hooks';
import { LayoutAnimation, UIManager } from 'react-native';
import { useSelector } from 'react-redux';

import renderHookWithProviders from 'utils/testHelpers/renderHookWithProviders';

import useContainer, { ANIMATION_DURATION } from '../hook';

jest.mock('react-native', () => {
  const RN = jest.requireActual('react-native');

  RN.LayoutAnimation.configureNext = jest.fn();
  RN.UIManager.setLayoutAnimationEnabledExperimental = jest.fn();

  return RN;
});

describe('FlashRoot useContainer hook tests', () => {
  useSelector.mockReturnValue([
    { id: '1', name: 'flash_1' },
    { id: '2', name: 'flash_2' },
  ]);

  let { result } = renderHookWithProviders(useContainer);

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  it('checks doAnimation method', () => {
    LayoutAnimation.configureNext.mockReset();

    act(() => {
      result.current.doAnimation();
    });

    expect(LayoutAnimation.configureNext).toHaveBeenCalledTimes(2);

    expect(LayoutAnimation.configureNext).toHaveBeenNthCalledWith(1, {
      duration: ANIMATION_DURATION,
      update: {
        type: LayoutAnimation.Types.linear,
      },
    });

    expect(LayoutAnimation.configureNext).toHaveBeenNthCalledWith(2, {
      duration: ANIMATION_DURATION,
      create: {
        type: LayoutAnimation.Types.linear,
        property: LayoutAnimation.Properties.scaleXY,
      },
    });
  });

  describe('Lifecycle', () => {
    it('should call setLayoutAnimationEnabledExperimental', () => {
      expect(UIManager.setLayoutAnimationEnabledExperimental).toHaveBeenCalledWith(true);
    });

    it('should not call setLayoutAnimationEnabledExperimental', () => {
      UIManager.setLayoutAnimationEnabledExperimental = null;

      ({ result } = renderHookWithProviders(useContainer));

      expect(UIManager.setLayoutAnimationEnabledExperimental).toBeFalsy();
    });
  });
});
