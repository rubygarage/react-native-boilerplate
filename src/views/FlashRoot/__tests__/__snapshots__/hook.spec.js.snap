// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashRoot useContainer hook matches snapshot 1`] = `
Object {
  "didMount": [Function],
  "doAnimation": [Function],
}
`;
